File: socialview-commons/src/main/java/com/hendraanggrian/widget/socialview/commons/HashtagAdapter.java
package com.hendraanggrian.widget.socialview.commons;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Filter;
import android.widget.TextView;

import java.text.NumberFormat;
import java.util.Locale;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
public final class HashtagAdapter extends SocialAdapter<Hashtag> {

    public HashtagAdapter(@NonNull Context context) {
        super(context, R.layout.item_hashtag, R.id.textview_hashtag_value);
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        final ViewHolder holder;
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.item_hashtag, parent, false);
            holder = new ViewHolder(convertView);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

        final Hashtag item = getItem(position);
        if (item != null) {
            holder.textViewHashtag.setText(item.getHashtag());

            if (item.getCount() != null) {
                holder.textViewHashtagCount.setText(item.getCount() == 1
                        ? "1 post"
                        : String.format("%s posts", NumberFormat.getNumberInstance(Locale.US).format(item.getCount())));
                holder.textViewHashtagCount.setVisibility(View.VISIBLE);
            } else {
                holder.textViewHashtagCount.setVisibility(View.GONE);
            }
        }
        return convertView;
    }

    @NonNull
    @Override
    public Filter initializeFilter() {
        return new SuggestionFilter() {
            @Override
            public String getString(Hashtag item) {
                return item.getHashtag();
            }
        };
    }

    private static class ViewHolder {
        @NonNull private final TextView textViewHashtag;
        @NonNull private final TextView textViewHashtagCount;

        private ViewHolder(@NonNull View view) {
            textViewHashtag = (TextView) view.findViewById(R.id.textview_hashtag_value);
            textViewHashtagCount = (TextView) view.findViewById(R.id.textview_hashtag_count);
        }
    }
}

File: socialview-commons/src/com/hendraanggrian/widget/HashtagAdapter.kt
package com.hendraanggrian.widget

import android.content.Context
import android.support.annotation.PluralsRes
import android.view.LayoutInflater.from
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.hendraanggrian.socialview.commons.R

/**
 * Default adapter for displaying hashtag in [SocialAutoCompleteTextView].
 * Note that this adapter is completely optional, any adapter extending
 * [android.widget.ArrayAdapter] can be attached to [SocialAutoCompleteTextView].
 */
class HashtagAdapter @JvmOverloads constructor(
    context: Context,
    @PluralsRes private val countPlural: Int = R.plurals.posts
) : SocialAdapter<Hashtag>(
    context,
    R.layout.socialview_layout_hashtag,
    R.id.socialview_hashtag
) {

    override fun Hashtag.convertToString(): String = hashtag

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val holder: ViewHolder
        var _convertView = convertView
        when (_convertView) {
            null -> {
                _convertView = from(context)
                    .inflate(R.layout.socialview_layout_hashtag, parent, false)
                holder = ViewHolder(_convertView!!)
                _convertView.tag = holder
            }
            else -> holder = _convertView.tag as ViewHolder
        }
        getItem(position)?.let { hashtag ->
            holder.hashtagView.text = hashtag.hashtag
            hashtag.count?.let {
                holder.countView.text = context.resources.getQuantityString(countPlural, it)
            }
        }
        return _convertView
    }

    private class ViewHolder(itemView: View) {
        val hashtagView: TextView = itemView.findViewById(R.id.socialview_hashtag)
        val countView: TextView = itemView.findViewById(R.id.socialview_hashtag_count)
    }
}